com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/ScreenSpaceReflections.compute

```diff
@@ -59,20 +70,300 @@ TEXTURE2D_X(_DepthTexture);
+    float phi = 2.0 * PI * u.y;
+    float t1  = r * cos(phi);
+    float t2  = r * sin(phi);
+    float s  = 0.5 * (1.0 + viewToLocal[2].z);
+    t2 = (1.0 - s) * sqrt(1.0 - t1 * t1) + s * t2;
+
+    // Reproject onto hemisphere
+    localH = t1 * viewToLocal[0] + t2 * viewToLocal[1] + sqrt(max(0.0, 1.0 - t1 * t1 - t2 * t2)) * viewToLocal[2];
+
+    // Transform the normal back to the ellipsoid configuration
+    localH = normalize(float3(roughness * localH.x, roughness * localH.y, max(0.0, localH.z)));
+
+    VdotH = saturate(dot(localV, localH));
+}
+
+float Lambda_GGX(float roughness,
+                 float3 V)
+{
+    return 0.5 * (sqrt(1.0 + (Sq(roughness * V.x) + Sq(roughness * V.y)) / Sq(V.z)) - 1.0);
+}
+
+bool SampleGGX(float roughness_,
+               float3x3 localToWorld,
+               float3 V,
+               float fresnel0,
+               float2 inputSample,
+           out float3 outgoingDir,
+           out float value,
+           out float pdf,
+           out float fresnel)
+{
+    float roughness = clamp(roughness_, MIN_GGX_ROUGHNESS, MAX_GGX_ROUGHNESS);
+
+    float VdotH;
+    float3 localV, localH;
+    SampleGGXVisibleNormal(inputSample, V, localToWorld, roughness, localV, localH, VdotH);
+
+    // Compute the reflection direction
+    float3 localL = 2.0 * VdotH * localH - localV;
+    outgoingDir = mul(localL, localToWorld);
+
+    if (localL.z < 0.001)
+        return false;
+
+    float pdfNoGV = D_GGX(localH.z, roughness);
+    float lambdaVPlusOne = Lambda_GGX(roughness, localV) + 1.0;
+    pdf = pdfNoGV / lambdaVPlusOne;
+
+    float lambdaL = Lambda_GGX(roughness, localL);
+    fresnel = F_Schlick(fresnel0, VdotH);
+    value = fresnel * pdfNoGV / (lambdaVPlusOne + lambdaL);
+
+    if (pdf < 0.001)
+        return false;
+
+    return true;
+}
+
+float PerceptualRoughnessFade(float perceptualRoughness, float fadeRcpLength, float fadeEndTimesRcpLength)
+{
+    float t = Remap10(perceptualRoughness, fadeRcpLength, fadeEndTimesRcpLength);
+    return Smoothstep01(t);
+}
+
+void GetHitInfos(uint2 positionSS, out float2 hitPositionNDC, out float srcPerceptualRoughness, out float3 positionWS, out float pdf, out float weight, out float3 N, out float3 L, out float3 V, out float dist, out float NdotL, out float NdotH, out float VdotH, out float NdotV)
+{
+    float2 uv = float2(positionSS) * _RTHandleScale.xy;
+
+    float4 hitData = _SsrHitPointTexture[COORD_TEXTURE2D_X(positionSS)];
+    dist = hitData.z;
+    hitPositionNDC = hitData.xy;
+
+    int globalSampleIndex = _FrameCount;
+    float2 Xi;
+    Xi.x = GetBNDSequenceSample(positionSS, globalSampleIndex, 0);
+    Xi.y = GetBNDSequenceSample(positionSS, globalSampleIndex, 1);
+
+    // Pick which subpixel we will be launching our effects from
+    float subPixelSample = GetBNDSequenceSample(positionSS, globalSampleIndex, 3);
+    int subPixel = clamp((int)(subPixelSample * 4.0), 0, 3);
+    uint2 shift = HalfResIndexToCoordinateShift[subPixel];
```

Comment:
This is code in RayTracingReflection.compute is meant to handle BSDF data info fetch in half res mode, in the current state SSR doesn't have half res mode. This is a copy paste mistake i guess.

Reply:
We switched to an algorithm that does temporal accumulation this shift help for quality. I am open to discuss that.

Reply:
This usage of the code, if on purpose, increases the variance and biases the signal even more, I am sure of that.

Reply:
That help on the edges & very close hit (corners: Wall vs Floor/Ceilling)